### DISCUSSION TITLE
Is it possible to do longest match in PhraseMatcher?

### DISCUSSION THREAD

**Discussion by lingvisa** [explosion/spaCy#7301] [New Features & Project Ideas]
`matcher = PhraseMatcher(nlp.vocab, attr = 'LOWER')`

Is there one option to only return longest match? For example:

In dictionary:

```
a test
test
```
And the incoming sentence is:

`This is a test.
`
The current phrase matcher returns both 'test' and 'a test' as matched offsets. May I only get the longest match 'a test' as the returned match, instead of both?

**Comment by svlandeg** [comment]
I don't think there is a direct way to do this with the `PhraseMatcher`, but it should be relatively straightforward to implement a custom fuction that will run through the `PhraseMatcher` results and filter them accordingly.

The `Matcher` does have a setting called `greedy` when adding a pattern, which you can set to "LONGEST" to match for instance the longest possible match when you have a pattern with optional or potentially repetitive parts.

**Comment by lingvisa** [comment]
Thanks. Longest match should be the behavior for lexicon entry matching.

**Comment by svlandeg** [comment]
(Maintenance note: this question is well suited for the discussions forum, so I'm moving it there. While it looks like this issue gets closed, in fact the linked discussion thread will remain open to continue the conversation)
